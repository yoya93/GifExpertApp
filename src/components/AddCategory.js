import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [imputValue, setImputValue] = useState("");

  const handleImputChange = (e) => {
    setImputValue(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();

    if (imputValue.trim().length > 2) {
      setCategories((cate) => {
        setCategories([imputValue, ...cate]);
        setImputValue("");
      });
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <p>{imputValue}</p>
      <input type="text" value={imputValue} onChange={handleImputChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
