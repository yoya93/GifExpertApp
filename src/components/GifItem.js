import React from "react";
import PropTypes from "prop-types";

export const GifItem = ({ title, url }) => {
  return (
    <div className="card  animate__heartBeat">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

// PrimeraApp.defaultProps = {
//   title: "titulo",
//   url: "https://youtube.com",
// };
