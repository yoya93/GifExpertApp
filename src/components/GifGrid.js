import React from "react";
import { GifItem } from "./GifItem";
import { useFetchGifs } from "./hooks/useFetchGifs";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h2 className="animate__wobble">{category}</h2>

      {loading && <p> Loading</p>}

      <div className="card-grid">
        {images.map((img) => {
          return <GifItem key={img.id} {...img} />;
        })}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
