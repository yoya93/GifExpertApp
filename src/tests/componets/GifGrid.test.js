import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { useFetchGifs } from "../../components/hooks/useFetchGifs";
jest.mock("../../components/hooks/useFetchGifs");

import { GifGrid } from "../../components/GifGrid";

describe("Pruebas en <GifGrid /> ", () => {
  test("debe mostrarse correctamente ", () => {
    const category = "Neymar jr";

    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar los items cuando se cargan lasimagenes con useFetchGifs   ", () => {
    const category = "Neymar jr";

    const gifs = [
      {
        id: "1",
        url: "https",
        title: "cuno",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifItem").length).toBe(gifs.length);
  });
});
