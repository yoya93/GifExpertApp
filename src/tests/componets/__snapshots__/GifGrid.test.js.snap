// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid />  debe de mostrar los items cuando se cargan lasimagenes con useFetchGifs    1`] = `
<Fragment>
  <h2
    className="animate__wobble"
  >
    Neymar jr
  </h2>
  <div
    className="card-grid"
  >
    <GifItem
      id="1"
      key="1"
      title="cuno"
      url="https"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid />  debe mostrarse correctamente  1`] = `
<Fragment>
  <h2
    className="animate__wobble"
  >
    Neymar jr
  </h2>
  <p>
     Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
